yuniql-platforms/oracle/OracleDataService.cs

```diff
@@ -0,0 +1,194 @@
+	applied_on_utc TIMESTAMP NOT NULL,
+	applied_by_user VARCHAR2(32) NOT NULL,
+	applied_by_tool VARCHAR2(32) NOT NULL,
+	applied_by_tool_version VARCHAR2(16) NOT NULL,
+    status VARCHAR2(32) NOT NULL,
+    duration_ms NUMBER NOT NULL,
+    checksum VARCHAR2(64) NOT NULL,
+    failed_script_path VARCHAR2(4000) NULL,
+    failed_script_error VARCHAR2(4000) NULL,
+    additional_artifacts VARCHAR2(4000) NULL,
+    CONSTRAINT pk_${YUNIQL_TABLE_NAME} PRIMARY KEY (sequence_id),
+	CONSTRAINT ix_${YUNIQL_TABLE_NAME} UNIQUE (version)
+);
+            ";
+
+        ///<inheritdoc/>
+        public string GetSqlForGetCurrentVersion()
+            => @"
+SELECT version FROM ${YUNIQL_TABLE_NAME} WHERE status = 'Successful' AND ROWNUM = 1 ORDER BY sequence_id DESC;
+            ";
+
+        ///<inheritdoc/>
+        public string GetSqlForGetAllVersions()
+            => @"
+SELECT sequence_id, version, applied_on_utc, applied_by_user, applied_by_tool, applied_by_tool_version, status, duration_ms, checksum, failed_script_path, failed_script_error, additional_artifacts
+FROM ${YUNIQL_TABLE_NAME} ORDER BY version ASC;
+            ";
+
+        ///<inheritdoc/>
+        public string GetSqlForInsertVersion()
+            => @"
+INSERT INTO ${YUNIQL_TABLE_NAME} (version, applied_on_utc, applied_by_user, applied_by_tool, applied_by_tool_version, status, duration_ms, checksum, failed_script_path, failed_script_error, additional_artifacts) 
+VALUES ('${YUNIQL_VERSION}', SYSDATE, USER, '${YUNIQL_APPLIED_BY_TOOL}', '${YUNIQL_APPLIED_BY_TOOL_VERSION}','${YUNIQL_STATUS}', '${YUNIQL_DURATION_MS}', '${YUNIQL_CHECKSUM}', '${YUNIQL_FAILED_SCRIPT_PATH}', '${YUNIQL_FAILED_SCRIPT_ERROR}', '${YUNIQL_ADDITIONAL_ARTIFACTS}');
+            ";
+
+        ///<inheritdoc/>
+        public string GetSqlForUpdateVersion()
+            => @"
+UPDATE ${YUNIQL_TABLE_NAME}
+SET
+    applied_on_utc          =  SYSDATE,
+    applied_by_user         =  USER,
+    applied_by_tool         = '${YUNIQL_APPLIED_BY_TOOL}', 
+    applied_by_tool_version = '${YUNIQL_APPLIED_BY_TOOL_VERSION}', 
+    status                  = '${YUNIQL_STATUS}', 
+    duration_ms             = '${YUNIQL_DURATION_MS}', 
+    failed_script_path      = '${YUNIQL_FAILED_SCRIPT_PATH}', 
+    failed_script_error     = '${YUNIQL_FAILED_SCRIPT_ERROR}', 
+    additional_artifacts    = '${YUNIQL_ADDITIONAL_ARTIFACTS}'
+WHERE
+    version                 = '${YUNIQL_VERSION}';
+            ";
+
+        ///<inheritdoc/>
+        public string GetSqlForUpsertVersion()
+            => throw new NotSupportedException("Not supported for the target platform");
+
+        ///<inheritdoc/>
+        public string GetSqlForCheckRequireMetaSchemaUpgrade(string currentSchemaVersion)
+        //when table __yuniqldbversion exists, we need to upgrade from yuniql v1.0 to v1.1 version
+        => @"
+SELECT 'v1.1' FROM SYS.ALL_TABLES WHERE OWNER = '${YUNIQL_DB_NAME}' AND TABLE_NAME = '__yuniqldbversion' AND ROWNUM = 1;
```

Comment:
Its not applicable here as its new platform, we can just do this here
```
        public string GetSqlForCheckRequireMetaSchemaUpgrade(string currentSchemaVersion)
            => @"
SELECT NULL;
            ";
```

Reply:
On it, followed exact implementation and replaced it with Oracle specific details.

Reply:
is this some sort of convention to return ```SELECT NULL;```? Or is it just dummy SQL execution as this can be converted to ```SELECT 1 FROM DUAL``` etc